File: Introduccion/ejercicio 2/Producto.java
package introduccion;

public class Producto {
    private String nombre;
    private double precio;
    private int stock;
    
    public Producto(String nombre, double precio, int stock){
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }
    
    public void vender( int cantidad){
        if ( cantidad <= stock){
            stock = stock - cantidad;
            System.out.println("Se vendieron "+ cantidad + " unidades de "+ nombre);
        }else{
            System.out.println("No hay suficiente stock para vender "+ cantidad+ " unidades.");
        }
    }
    
    public void reabastecer (int cantidad){
        stock = stock + cantidad;
        System.out.println("Se reabastecieron "+ cantidad +" unidades de "+nombre);
    }
    
    @Override 
    public String toString(){
        return"Producto: " +nombre+ ", Precio: Bs "+ precio +", stock disponible: "+ stock;
    }
}

File: Manejo de Excepciones/ejercicio 2/Producto.py
class Producto:
    def __init__(self, codigo, nombre, precio, stock):
        self.codigo = codigo
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def mostrar(self):
        print(f"{self.codigo} - {self.nombre} - ${self.precio} - Stock: {self.stock}")